File: core/src/main/java/y2k/joyreactor/Message.java
package y2k.joyreactor;

import rx.Observable;
import y2k.joyreactor.requests.MessageRequest;

import java.io.Serializable;
import java.util.Date;
import java.util.List;

/**
 * Created by y2k on 10/1/15.
 */
public class Message implements Serializable {

    public String text;
    public Date date;
    public boolean isMine;
    public String userName;

    public static Observable<List<Message>> request(String name) {
        return new MessageRequest().request(name);
    }
}

File: core/src/main/kotlin/y2k/joyreactor/Message.kt
package y2k.joyreactor

import y2k.joyreactor.services.repository.DataSet
import java.io.Serializable
import java.util.*

/**
 * Created by y2k on 10/1/15.
 */
class Message(
    val text: String,
    val date: Date,
    val isMine: Boolean,
    val userName: String,
    val userImage: String?) : DataSet.Dto, Serializable {

    override var id: Long = 0

    fun getUserImageObject(): UserImage {
        return if (userImage == null) UserImage() else UserImage(userImage)
    }
}